### PR TITLE
krita: 5.0.2 -> 5.0.6, fix python plugin and dependencies

### PR DESCRIPTION
**Pull request by sifmelcara**
###### Description of changes

Builtin python plugins (e.g. ten brushes) was not working because krita's python scripts cannot find `sip` during build. (Upstream changed how this works in recent versions, not sure when did this break).

`libmypaint` is for MyPaint brush engine introduced in krita 5. Builtin mypaint brushes won't work without this dependency.

`libwebp` is for exporting to WebP format.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by sifmelcara** [comment]
Hi @wackbyte @Kranzes , I wonder if you folks could help test this change? (I saw you folks helped update krita before)

Thanks!

**Comment by wackbyte** [comment]
MyPaint brushes and exporting to WebP both work.

Result of `nixpkgs-review pr 174101` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>gmic-qt-krita</li>
    <li>krita</li>
  </ul>
</details>

File: pkgs/applications/graphics/krita/generic.nix [hunk 1]
```diff
@@ -23,20 +23,30 @@ mkDerivation rec {
...
     ++ lib.optional stdenv.cc.isGNU "-Wno-deprecated-copy";
 
+  # Krita runs custom python scripts in CMake with custom PYTHONPATH which krita determined in their CMake script.
+  # Patch the PYTHONPATH out so python scripts can import sip successfully.
+  prePatch = let pythonPath = python3Packages.makePythonPath [ python3Packages.sip python3Packages.setuptools ];
+             in ''
+      substituteInPlace cmake/modules/FindSIP.cmake \
+              --replace 'PYTHONPATH=''${_sip_python_path}' 'PYTHONPATH=${pythonPath}'
+      substituteInPlace cmake/modules/SIPMacros.cmake \
+              --replace 'PYTHONPATH=''${_krita_python_path}' 'PYTHONPATH=${pythonPath}'
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/krita/generic.nix:47]:
```suggestion
  postPatch = let
    pythonPath = python3Packages.makePythonPath with python3Packages; [ sip setuptools ];
  in ''
    substituteInPlace cmake/modules/FindSIP.cmake \
      --replace 'PYTHONPATH=''${_sip_python_path}' 'PYTHONPATH=${pythonPath}'
    substituteInPlace cmake/modules/SIPMacros.cmake \
      --replace 'PYTHONPATH=''${_krita_python_path}' 'PYTHONPATH=${pythonPath}'
```

Comment (sifmelcara) [reply pkgs/applications/graphics/krita/generic.nix]: Done. Thanks!

**Pull request merged**